atc0005/check-cert
# Initial implementation of `Root` cert validation

## Overview

This set of changes applies to both the `check_cert` plugin and the `lscert` CLI tool. These changes borrow heavily from recent work for the initial implementation of the `Chain Order` validation check. As with that implementation, this set of changes is subject to change as work to implement further validity checks (and refactor existing ones) continues.

## Changes

- update `check_cert` plugin
  - add new `Root` validation type
    - assert that no root certificates are in the chain
    - remove 'note' from the `Chain Order` validation check when a root certificate is found (now handled by this check)
  - extend tests to cover new validation type
- update `lscert`
  - incorporate new validation check
  - incorporate the same "advice" output that the `check_cert` plugin now emits for `Root` validation problems
- shared
  - move "advice" helper funcs to separate validation-helpers file as refactoring step towards further upcoming validation checks work
    - the initial idea was to provide further advice for sysadmins when a root certificate was detected (this may still occur later)
- update README to add coverage for new `Root` validation check
  - emphasize that this check is not applied by default, but that this could change in the future
  - sysadmins are encouraged to explicitly opt-out of validation checks that they are not interested in
    - the checks still run, but the results are marked as ignored and not used to trigger plugin state changes

## References

- GH-1004
- GH-364
- GH-365
